cluster/charts/kustomize_patches.yaml

```diff
@@ -0,0 +1,11 @@
+# We need InfrastructurePublications to be deleted first because if two resources
+# with pre-delete hooks have the same hook weight, they are deleted in alphabetic
+# order. So, Helm tries to delete InfrastructureDefinitions but they cannot be
+# deleted if they are published. Hence publications need to be deleted first.
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    "helm.sh/hook-weight": "-1"
```

Comment:
`InfrastructurePublication`s need to be deleted before the corresponding definitions because definitions are locked by a publication via finalizer. @negz we got some thoughts around this dependency like either making publication a field on definition or remove the dependency completely but they are not yet implemented.